### ISSUE TITLE
Cannot download artifacts until all flows in a workflow have finished running

### ISSUE THREAD

**Issue by ezyang** [pytorch/test-infra#3952]
This seems to be a GHA limitation

**Comment by malfet** [comment]
Yes, indeed it is, though we probably can workaround that by mirroring the artifacts to s3

**Comment by clee2000** [comment]
Artifacts are usually uploaded to s3 now, which can be downloaded whenever

I also think GH changed the behavior so you can download even if some jobs in the workflow aren't finished
